internal/controller/nnf_node_block_storage_controller.go

```diff
@@ -644,7 +684,275 @@ func (r *NnfNodeBlockStorageReconciler) getStorageGroup(ss nnf.StorageServiceApi
 }
 
 func (r *NnfNodeBlockStorageReconciler) deleteStorageGroup(ss nnf.StorageServiceApi, id string) error {
-	return ss.StorageServiceIdStorageGroupIdDelete(ss.Id(), id)
+	err := ss.StorageServiceIdStorageGroupIdDelete(ss.Id(), id)
+	if err != nil {
+		ecErr, ok := err.(*ec.ControllerError)
+		// If the error is from a 404 error, treat as success (idempotent deletion)
+		if ok && ecErr.StatusCode() == http.StatusNotFound {
+			return nil
+		}
+	}
+	return err
+}
+
+// getComputeNodesWithAccess returns a list of compute node names that have access to this NnfNodeBlockStorage
+func (r *NnfNodeBlockStorageReconciler) getComputeNodesWithAccess(blockStorage *nnfv1alpha9.NnfNodeBlockStorage) []string {
+	computeNodes := make(map[string]bool)
```

Comment:
Does this need to be put into a map first? Is that to remove duplicates? Maybe add a comment if that's the case.